### ISSUE TITLE
Consistent crash when LSPString::get_native() returns null

### ISSUE THREAD

**Issue by falkTX** [sadko4u/lsp-plugins#185]
Trying out LSP IR plugins, I am having a crash as soon as I try to load a file.
Debugging a little, I see the crash is due to this line in the AudioFile.cpp file:
```
status_t res = fd.open(path->get_native());
```

On the system I am testing with (an embed linux device) `LSPString::get_native()` returns null.

The very first lines of the function is where it returns null, more specifically, this:
```
        // Open conversion
        iconv_t cd = init_iconv_from_wchar_t(charset);
        if (cd == iconv_t(-1))
            return NULL;
```

**Comment by falkTX** [comment]
I tried changing the default null charset to be "UTF-8", but still have a crash.
Changing the line:
```
iconv_t cd = init_iconv_from_wchar_t(charset);
```
to this:
```
iconv_t cd = iconv_open("UTF-8", "WCHAR_T");
```
no longer returns null, and thus does not crash.


btw I also tried this:
```
iconv_t cd = iconv_open(charset, "WCHAR_T");
```
but that also crashes.

**Comment by sadko4u** [comment]
Hello Filipe!

If charset is NULL, the LSPString tries to obtain it from the LC_CTYPE locale settings variable:
https://github.com/sadko4u/lsp-plugins/blob/663223c8075c3b4476f06c7d712c2716cec9daa5/src/core/io/charset.cpp#L517-L553

What's your system LC_CTYPE value?

**Comment by sadko4u** [comment]
Anyway, fd.open should not crash if argument is NULL, I need to check, thanks.

**Comment by falkTX** [comment]
> What's your system LC_CTYPE value?

There is no such variable.
And even then, the host runs as part of a systemd service, which strips away pretty much all environment variables.

I can reproduce the crash from an ssh session, for that the full list of env vars is:
```
# env | sort
_=/bin/env
EDITOR=/bin/vi
HISTFILESIZE=1000
HISTSIZE=1000
HOME=/root
HOSTNAME=modduox
INPUTRC=/etc/inputrc
LANG=en_US.UTF-8
LOGNAME=root
LS_COLORS=no=00:fi=00:di=01;34:ln=01;36:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arj=01;31:*.taz=01;31:*.lzh=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.gz=01;31:*.bz2=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.jpg=01;35:*.jpeg=01;35:*.png=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.mpg=01;35:*.mpeg=01;35:*.avi=01;35:*.fli=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:
MAIL=/var/mail/root
PAGER=/bin/more 
PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin
PS1=[\u@\h \W]\$ 
PWD=/root
SHELL=/bin/sh
SHLVL=1
TERM=xterm-color
USER=root
```

**Comment by sadko4u** [comment]
@falkTX what does
```
locale
```
show?

**Comment by sadko4u** [comment]
Additionally, what does ```iconv -l``` show?

**Comment by falkTX** [comment]
There are no such tools on the system.
```
[root@modduox ~]# locale
-sh: locale: command not found
[root@modduox ~]# iconv
-sh: iconv: command not found
```
this is intentionally a very barebones linux system, one of those "just enough OS" type things, as built by buildroot, yocto and others.

**Comment by sadko4u** [comment]
Uh... I never dealt with Linux systems with such stripped locale settings. How does the other software detect the locale it's using?

Anyway, I've committed some changes to the **github-issue-185** branch.

Please verify the build from this branch.

**Comment by falkTX** [comment]
This is the first ever software I see crashing on the device because of the locale. Even earlier versions of LSP worked fine.
To help applications, I specifically set `LANG=en_US.UTF-8` as env var in the host.
tbh it surprises me that a plugin crashes because of this.

I will test that branch, thanks.

**Comment by falkTX** [comment]
hmm are you sure that we are meant to use `iconv_open("WCHAR_T", "UTF-8")` and not `iconv_open("UTF-8", "WCHAR_T")` ?

**Comment by sadko4u** [comment]
@falkTX there are two initialization functions: from 32-bit char to native and back. I suppose you're looking the direct version.

**Comment by falkTX** [comment]
Just tested, files are loaded without a crash now.
Thanks for the quick fix!

**Comment by sadko4u** [comment]
Thanks for the report.
WIll be available in 1.1.30.

**Comment by falkTX** [comment]
Cool cool. I forgot how you managed the tickets, but in any case, feel free to close this ticket now.

**Comment by sadko4u** [comment]
I'll close it after the 1.1.30 release. Don't worry about the status.

**Comment by sadko4u** [comment]
Merged changes into **devel** branch: d364b736b8cd3640f232bb8f4208d7cc89d5cad2

**Comment by sadko4u** [comment]
Removed the **github-issue-185** branch.

**Comment by sadko4u** [comment]
Available in 1.1.30
